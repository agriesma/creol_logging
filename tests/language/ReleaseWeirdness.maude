load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Weirdness" : Class | Inh: noInh, Param: noVid, Att: "c" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "c" @ "Weirdness" ) ; bool(false) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "m" : Method | Param: noVid, Att: "_" |-> null, Code: release ;
              await "c" ; return ( emp ) >,
            < "n" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( "c" ; bool(true) ) ; return ( emp ) >,
      Ocnt: 0 >
    
    < "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab1" |-> null,
              "ccglab0" |-> null, "w" |-> null, "_" |-> null,
              Code: new( "w" ; "Weirdness" ; emp ) ; call( "ccglab0" ; "w" ;
              "m" ; emp ) ; free( "ccglab0" ) ; call( "ccglab1" ; "w" ; "n" ;
              emp ) ; free( "ccglab1" ) ; return ( emp ) >,
      Ocnt: 0 > } .
endm

