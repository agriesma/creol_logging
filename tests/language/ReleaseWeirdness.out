==========================================
reduce in PROGRAM : state .
result State: {
< "Test" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "run" : Method | Param: noVid, Att: "_" |-> null, "ccglab0" |-> null,
    "ccglab1" |-> null, "w" |-> null, Code: new("w" ; "Weirdness" ; emp) ;
    call("ccglab0" ; "w" ; "m" ; emp) ; free("ccglab0") ; call("ccglab1" ; "w"
    ; "n" ; emp) ; free("ccglab1") ; return(emp) >,
  Ocnt: 0 >
 
< "Weirdness" : Class | Inh: noInh, Param: noVid, Att: "c" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("c" @
    "Weirdness" ; bool(false)) ; return(emp) >, < "m" : Method | Param: noVid,
    Att: "_" |-> null, Code: release ; await "c" ; return(emp) >, < "n" :
    Method | Param: noVid, Att: "_" |-> null, Code: assign("c" ; bool(true)) ;
    return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null, Code:
    skip ; return(emp) >,
  Ocnt: 0 >
}
Bye.
